### PR TITLE
Feature/data integrity test

### PR DESCRIPTION
**Pull request by kfang4**
To note, there are some updates in the `feature/update-manifest` branch that haven't been implemented yet since the pipeline isn't being rerun yet, which means this script will need to be updated when that time comes.

This script will generate a Markdown report for each team and an output log. To test:
- One team: `util/promote_staging_data -t team-lee`
- Multiple teams: `util/promote_staging_data -t team-lee,team-jakobsson`

### CODE REVIEW

**Comment by hkeward** [comment]
@kfang4 can you post a screenshot of an example report (paths can be made up, just want to see the format)

**Review by hkeward — COMMENTED** [review]
Looks good, a few small comments:
- I like having file checksums output for modified files -- it feels like we should have them for everything; I think getting them into the manifest itself (and the metadata files, generated upon file upload) would be really nice
- I think it could be a bit clearer which set of files / info is being referred to by referencing it as "current (ie the data that this file sits alongside)" and "previous (whatever we clobbered to make the files you're looking at)" rather than "staging" and "prod" or "initial" and "target"
- I think table 3 should either come before the 'files changed, modified, deleted' and 'data integrity tests results' sections (so you understand definitions before reading through those sections), or the info there should be integrated into those sections
- It seems like this report could reference the manifest files without including their full contents, and that would make it a bit cleaner (and more machine-readable) - if that's the case, we should also preserve manifests over time

I wonder if reports and manifests should live side-by-side in some other path like `metadata` or `file_info` or something like that.

I'm thinking something like (not necessarily using these names, just as an example):

```
.
├── cohort_analysis
├── metadata
│   ├── 2024-06-15T16-04-37Z
│   │   ├── MANIFEST.tsv # all of the files from preprocessing and cohort
│   │   └── report.md
│   └── 2024-07-25T15-55-32Z
│       ├── MANIFEST.tsv
│       └── report.md
└── preprocessing
```

So the order of operations would be:
1. Test data
2. Write report locally; include paths to the previous and current manifests (even if you haven't yet written the current manifest to that loc)
3. Promote data
4. Copy report and manifest to the `metadata/${timestamp}` directory.

WDYT?

**Comment by kfang4** [comment]
All comments make sense- will implement these changes. One thing, the order operations would instead be:
1. Test data
2. Write report locally; include paths to the previous and current manifests (even if you haven't yet written the current manifest to that loc)
3. Copy report and manifest to the metadata/${timestamp} directory. 
4. Promote data

If data is promoted, the prod manifests will be overwritten.

**Comment by kfang4** [comment]
To note:
- Changes in `feature/update-manifest` include renaming workflow to _just_ harmonzied_pmdbs rather than preprocess and cohort_analysis in manifest files
- Changes in `feature/update-manifest` include adding hash (md5) in manifest files
- Changes in `feature/update-manifest` include adding the GitHub release link in manifest files
- "Previous manifest gs location" referred in this script does not exist yet

**Comment by kfang4** [comment]
Dave (MJFF) reviewed report.

File: util/promote_staging_data [hunk 1]
```diff
@@ -452,27 +464,36 @@ log "Data integrity test results for all teams:"
 echo -e "team\tall_tests_passed"
 echo -e "${all_teams_results}"
 
-# Try syncing staging data to production
 while read -r team result_status; do
   if [[ "$result_status" == True ]]; then
     staging_bucket=gs://asap-${STAGING_BUCKET_TYPE}-data-${team}
     production_bucket=gs://asap-curated-data-${team}
```
Comment (hkeward) [inline util/promote_staging_data:470]: Will we ever go DEV -> UAT?

**Review by hkeward — COMMENTED** [review]
LGTM, pending approval on the report structure

File: util/promote_staging_data (hunk 1 above)
Comment (kfang4) [reply util/promote_staging_data:470]: @hkeward This was part of the original script you wrote. AFAIK we haven't encountered this yet.

**Comment by kfang4** [comment]
@hkeward Added a description for data integrity test result tables as per Dave's request. Merging.

**Pull request merged**